lib/utils.js

```diff
@@ -31,48 +32,46 @@ function normalizeBinary (binaryPath, platform, arch) {
                platform;
 
-    var channelNames = ["firefox", "firefoxdeveloperedition", "beta", "nightly", "aurora"];
-
     var app = binaryPath.toLowerCase();
-    var result = null;
-    if (platform === "osx" && channelNames.indexOf(binaryPath) !== -1) {
-      // Try to find the app for this channel.
-      // mdfind "kMDItemCFBundleIdentifier == 'org.mozilla.firefoxdeveloperedition'"
-      var results = spawnSync("mdfind", ["kMDItemCFBundleIdentifier == 'org.mozilla." + binaryPath + "'"]);
-      if (results.stdout) {
-        result = results.stdout.toString().split('\n')[0];
+
+    if (platform === "osx") {
+      var result = null;
+      var channelNames = ["firefox", "firefoxdeveloperedition", "beta", "nightly", "aurora"];
+
+      if (channelNames.indexOf(binaryPath) !== -1) {
+        // Try to find the app for this channel.
+        // mdfind "kMDItemCFBundleIdentifier == 'org.mozilla.firefoxdeveloperedition'"
+        var results = spawnSync("mdfind", ["kMDItemCFBundleIdentifier == 'org.mozilla." + binaryPath + "'"]);
+        if (results.stdout) {
+          result = results.stdout.toString().split('\n')[0];
+        }
       }
-    }
-    binaryPath = result ||
-                 normalizeBinary.paths[app + " on " + platform] ||
-                 normalizeBinary.paths[app + " on " + platform + arch] ||
-                 binaryPath;
+      binaryPath = result ||
+                   normalizeBinary.paths[app + " on " + platform] ||
+                   binaryPath;
+      var isAppPath = path.extname(binaryPath) === ".app";
 
-    var isAppPath = platform === "osx" && path.extname(binaryPath) === ".app";
+      // On OSX, if given the app path, resolve to the actual binary
+      binaryPath = isAppPath ? path.join(binaryPath, "Contents/MacOS/firefox-bin") :
+                   binaryPath;
 
-    // On OSX, if given the app path, resolve to the actual binary
-    binaryPath = isAppPath ? path.join(binaryPath, "Contents/MacOS/firefox-bin") :
-                 binaryPath;
-    // not Windows or binary path given
-    if (platform.indexOf("windows") === -1 || channelNames.indexOf(app) === -1) {
       return resolve(binaryPath);
     }
-    // Windows binary finding
-    var appName = "Mozilla Firefox";
-    switch(app) {
-      case "beta":
-        // the default path in the beta installer is the same as the stable one
-        appName = "Mozilla Firefox";
-        break;
-      case "aurora":
-        appName = "Aurora";
-        break;
-      case "nightly":
-        appName = "Nightly";
-        break;
-      default:
-        break;
+    // Return the path if it contains at least two segments
+    else if (binaryPath.indexOf(path.sep) !== -1) {
+      return resolve(binaryPath);
+    }
+    // On linux but no path yet, use which to try and find the binary
+    else if (platform.indexOf("linux") !== -1) {
+        binaryPath = normalizeBinary.appNames[binaryPath + " on linux"] || binaryPath;
```

Comment:
oh, I didn't realize this had merged. It's not a big deal so don't worry about it 👍